bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BXMLItem.java

```diff
@@ -890,4 +891,42 @@ public synchronized void attemptFreeze(BVM.FreezeStatus freezeStatus) {
             this.freezeStatus = freezeStatus;
         }
     }
+
+    private static class BXmlAttrMap extends BMap {
+        private final BXMLItem bXmlItem;
+        private boolean constructed = false;
+
+        BXmlAttrMap(BXMLItem bXmlItem) {
+            super(new BMapType(BTypes.typeString));
+            this.bXmlItem = bXmlItem;
+        }
+
+        void finishConstruction() {
+            constructed = true;
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public void put(Object key, BValue value) {
+            super.put(key, value);
+            if (constructed) {
```

Comment:
What's the purpose of this check?

Reply:
This maps put method work under 2 states, 
1. While being populated with attribute values, using this put method.
2. Assigning new attributes to the XMLItem via this attribute map instance. (constructed is true)